src/runtime.rs

```diff
@@ -30,15 +29,20 @@ impl OCamlRuntime {
     ///
     /// After the first invocation, this method does nothing.
     pub fn init_persistent() {
-        static INIT: Once = Once::new();
-
-        INIT.call_once(|| {
-            let arg0 = "ocaml\0".as_ptr() as *const ocaml_sys::Char;
-            let c_args = vec![arg0, core::ptr::null()];
-            unsafe {
-                caml_startup(c_args.as_ptr());
-            }
-        })
+        #[cfg(not(feature = "no-caml-startup"))]
+        {
+            static INIT: std::sync::Once = std::sync::Once::new();
```

Comment:
Shouldn't this function always call `caml_startup`? As written, calling `init` above, allowing its result to be dropped (calling `caml_shutdown`) and then calling `init` again will return an `OCamlRuntime` not backed by an available runtime.

`caml_startup` will already avoid re-initialising if it is already running (and bumps a counter, to allow a corresponding `caml_shutdown` to behave non-destructively), so I believe it should be safe and more correct to simply delete this.

Reply:
It should **never** be called from an OCaml program that wants to use Rust code, hence the panic. You only do this initialization once in your program, from your `main` function or somewhere close to it (you may for example have a separate thread that has exclusive access to the OCaml runtime, in that case you would do it there). But no function exposed to OCaml should be doing this.

Reply:
Remember that this is nothing else than a workaround for the `dune utop` issue, the panic is in there just as a safeguard (to help anyone misusing the library), but this shouldn't be called in such case.

Reply:
I don't think this is enforced anywhere: you could write
```rust
fn call_first() {
  let r1 = init();
  ...
}
fn call_snd() {
  let r2 = init();
  ...
}
pub fn main() {
  call_first();
  call_snd();
}
```
and the second call won't get an OCaml runtime. I suppose this can be a separate issue.